src/internal/middleware.js

```diff
@@ -91,5 +93,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     return task
   }
 
+  sagaMiddleware.setContext = (props) => {
+    // TODO: or maybe `setContext` should be available on task and this one should act only as proxy method?
+    // would that be useful?
+    // const task = yield takeEvery(...)
+    // const task.setContext({...}) or maybe even - yield setContext(task, {...})?
```

Comment:
the only thing im not so happy about is that `sagaMiddleware` is kinda superior to the `runSaga` here, as the latter do not have possibility to share such things in an elegant way, but I guess one can always write some wrapper function up to his needs